kv/kvserver: TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer failed

kv/kvserver.TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer [failed](https://tanzanite.cluster.engflow.com/invocations/default/906cf9a3-8867-4b33-baad-11688325c9f9?testReportRun=6&testReportShard=33&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.1 @ [ab04f621a4c834c9f0cc7fd65187c531553ce384](https://github.com/cockroachdb/cockroach/commits/ab04f621a4c834c9f0cc7fd65187c531553ce384):


```
=== RUN   TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer2042928710
    test_log_scope.go:81: use -show-logs to present logs inline
    client_replica_test.go:865: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_replica_test.go:865
        	Error:      	Should be true
        	Test:       	TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer2042928710
--- FAIL: TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer (0.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=33</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42235

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -863,9 +863,14 @@ func TestTxnReadWithinUncertaintyIntervalAfterLeaseTransfer(t *testing.T) {
 	// The transaction has a read timestamp beneath the write's commit timestamp
 	// but a global uncertainty limit above the write's commit timestamp. The
 	// observed timestamp collected is also beneath the write's commit timestamp.
-	require.True(t, txn.ReadTimestamp.Less(writeTs))
-	require.True(t, writeTs.Less(txn.GlobalUncertaintyLimit))
-	require.True(t, txn.ObservedTimestamps[0].Timestamp.ToTimestamp().Less(writeTs))
+	assert.True(t, txn.ReadTimestamp.Less(writeTs))
+	assert.True(t, writeTs.Less(txn.GlobalUncertaintyLimit))
+	assert.True(t, txn.ObservedTimestamps[0].Timestamp.ToTimestamp().Less(writeTs))
+
+	if t.Failed() {
+		t.Logf("writeTs=%s, txn=%+v", writeTs, txn)
+		t.FailNow()
+	}
 
 	// Add a replica for key A's range to node 2. Transfer the lease.
 	tc.AddVotersOrFatal(t, keyA, tc.Target(1))

